### DISCUSSION TITLE
Composer Require/Composer Update wipes bundles EVERY SINGLE TIME

### DISCUSSION THREAD

**Discussion by brbintenyears** [pimcore/pimcore#10794] [General]
This is semi-expected behavior from composer, but not really.

In other frameworks, running Composer Update *will* wipe Vendor and reinstall all package, which is why it's bad practice to make any changes to vendor--it won't be there forever. 

With that said, in Pimcore, even ```Composer Require package/package-name``` wipes everything in the /bundles directory. And frankly this is ridiculous. All project assets that were manually imported there are wiped, and this should NOT happen.

**Comment by dpfaffenbauer** [comment]
Sorry what? I never had any issue with a composer update?!

**Comment by matthiashamacher** [comment]
Do you mean the public/bundles-Folder? That is also intended behavior, which comes from Symfony. It is because of the `bin/console assets:install` Command which runs after almost any composer execution as a post script, because otherwise you won't have the assets of the newly installed bundle.
If you don't want them to execute you can run composer with --no-script but that also means that you won't have the assets of that bundle if it provides some. And some other commands which are provided by pimcore will also not execute.

**Comment by jdreesen** [comment]
The `public/bundles`-Folder is just not the right place for "manually imported" assets. If you want assets there, place them in your bundle's `Resouces/public` folder, and they will automatically get copied via Symfony's `bin/console assets:install`-Command to this folder. This happens automatically on e.g. `composer require` or `composer update` in Pimcore.

If you really want "manually imported" assets, just place them in any other custom folder inside `public`.

**Reply by brbintenyears** [reply]
I was not aware of this. Coming from a Laravel background, I've always placed assets inside the /public directory with no issues. Thank you.
